### PR TITLE
[pre-commit] - add default template

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9024

## Description
Add default template config file

### CODE REVIEW

**Comment by JudahSchwartz** [comment]
make @anara123 and myself codeowners of this file

**Review by JudahSchwartz — CHANGES_REQUESTED** [review]
we should add a pipeline that checks this file on a sanity repo

**Comment by GuyAfik** [comment]
> we should add a pipeline that checks this file on a sanity repo

done

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -317,6 +317,14 @@ jobs:
           cd content
           demisto-sdk pre-commit --validate --show-diff-on-failure --verbose -i Packs/HelloWorld -i Packs/QRadar/Integrations/QRadar_v3 --mode=nightly
 
+      - name: run pre-commit with default pre-commit config shipped with the SDK
+        timeout-minutes: 60
+        run: |
+          source $(poetry env info --path)/bin/activate
+          cd content
```
Comment (JudahSchwartz) [inline .github/workflows/on-push.yml:324]: what repo is this?

File: .github/workflows/on-push.yml [hunk 2]
```diff
@@ -317,6 +317,14 @@ jobs:
...
+          rm -rf .pre-commit-config_template.yaml
+          demisto-sdk pre-commit --validate --show-diff-on-failure --verbose --mode=ci
```
Comment (JudahSchwartz) [inline .github/workflows/on-push.yml:326]: can we capture output, make sure some custom named hook gets run?

File: .github/workflows/on-push.yml (hunk 1 above)
Comment (GuyAfik) [reply .github/workflows/on-push.yml:324]: content repo after deleting the original .pre-commit-config_template.yaml

would like to test it out on some random repo?

File: .github/workflows/on-push.yml (hunk 2 above)
Comment (GuyAfik) [reply .github/workflows/on-push.yml:326]: which hooks for example?

File: .github/workflows/on-push.yml (hunk 1 above)
Comment (JudahSchwartz) [reply .github/workflows/on-push.yml:324]: i think this is fine unless Anar says otherwise

File: .github/workflows/on-push.yml (hunk 2 above)
Comment (JudahSchwartz) [reply .github/workflows/on-push.yml:326]: pytest or anything, just give a name so it doesnt run exit 0 silently without doing anything

File: demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    - -EnableExit
+    - -Severity
+    - Error
+    - -Path
+    files: .ps1$
+    exclude: .Tests.ps1$|test_data
+    copy_files:
+    - Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1
+    - Tests/demistomock/demistomock.ps1
+    run_isolated: true
```
Comment (GuyAfik) [inline demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml:125]: remove

File: demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    entry: pylint
+    files: Packs\/.*\.py$
+    exclude: _test\.py|.vulture_whitelist\.py|test_data
+    args:
+    - --ignore=demistomock.py,CommonServerPython.py,CommonServerUserPython.py,conftest.py,.venv
+    - -E
+    - --disable=bad-option-value,unsubscriptable-object
+    - -d duplicate-string-formatting-argument
+    - "--msg-template='{path}:{line}:{column}: {msg_id} {obj}: {msg}'"
+    - --generated-members=requests.packages.urllib3,requests.codes.ok
```
Comment (GuyAfik) [inline demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml:66]: remove

File: demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    min_py_version: '3.7'
+    args:
+    - --ignore-missing-imports
+    - --check-untyped-defs
+    - --show-error-codes
+    - --follow-imports=silent
+    - --allow-redefinition
+    exclude: test_data|tests_data|.venv|.*_test.py$|infrastructure_tests|.vulture_whitelist.py|demistomock.py|Templates|conftest.py
+    language: system
+    entry: mypy
```
Comment (GuyAfik) [inline demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml:50]: remove

File: .github/workflows/on-push.yml (hunk 2 above)
Comment (GuyAfik) [reply .github/workflows/on-push.yml:325]: can you explain what exactly do you want here?

**Review by JudahSchwartz — APPROVED** [review]

**Comment by JudahSchwartz** [comment]
just note the last 2 comments

File: .github/workflows/on-push.yml (hunk 1 above)
Comment (GuyAfik) [reply .github/workflows/on-push.yml:325]: keeping it as it is

**Pull request merged**